Tests/CouchDBTests/CouchDBTest.swift

```diff
@@ -52,62 +52,37 @@ class CouchDBTest: XCTestCase {
 
     var database: Database?
 
-    // MARK: - Initializers and test set-up and tear-down
-
-    override func setUp() {
-        delay(dropDatabaseIfExists)
-    }
-    
-    override func tearDown() {
-        delay(dropDatabaseIfExists)
-    }
-    
     /// Drop the test database, if it exists.
     ///
-    func dropDatabaseIfExists() {
-          // Check if DB exists
-          couchDBClient.dbExists(dbName) { exists, error in
-            if  error != nil {
-                XCTFail("Failed checking existence of database \(self.dbName). Error=\(error!.localizedDescription)")
-            } else {
-                if  exists {
-                    self.dropDatabase()
+    func dropDatabaseIfExists(completion: @escaping () -> Void) {
+        // Check if DB exists
+        DispatchQueue.global().asyncAfter(deadline: .now() + 1.0) {
+            self.couchDBClient.dbExists(self.dbName) { exists, error in
+                if  error != nil {
+                    XCTFail("Failed checking existence of database \(self.dbName). Error=\(error!.localizedDescription)")
+                } else {
+                    if  exists {
+                        self.dropDatabase(completion)
+                    }
                 }
             }
         }
     }
 
-    /// Create the test database, failing the test if it already exists, or if there
-    /// is a connectivity error.
-    ///
-    func createDatabase() {
-        delay(delayedCreateDatabase)
+    // setting up  the database
+    func setUpDatabase(isSetUpCompleted: @escaping () -> Void) -> Void {
+        self.dropDatabaseIfExists() {
+            self.createDatabase() {
```

Comment:
`createDatabase()` also needs a delay here, so that we do not exhaust the Cloudant API rate limit (currently the `setUpDatabase()` function will perform 2 API calls in 1 second).

Reply:
Sure will do ..